src/cc/syms.h

```diff
@@ -69,10 +69,24 @@ class KSyms : SymbolCache {
 };
 
 class ProcSyms : SymbolCache {
+  struct NameIdx {
+    size_t section_idx;
+    size_t str_table_idx;
+    size_t str_len;
+  };
+
   struct Symbol {
     Symbol(const std::string *name, uint64_t start, uint64_t size)
         : name(name), start(start), size(size) {}
-    const std::string *name;
+    Symbol(size_t section_idx, size_t str_table_idx, size_t str_len, uint64_t start,
+           uint64_t size)
+        : start(start), size(size) {
+      name_idx.section_idx = section_idx;
+      name_idx.str_table_idx = str_table_idx;
+      name_idx.str_len = str_len;
+    }
+    struct NameIdx name_idx;
```

Comment:
Maybe we can do
```
struct Symbol {
  uint32_t isset;
  union {
    std::string* name;
    struct NameIdx name_idx;
  } data;
};
So that we save little memory in either case?